### PR TITLE
Move JunitXML and run_test functions to their own files

### PR DESCRIPTION
**Pull request by jougs**
`do_tests.py` got way too complicated over time. The changes in this PR move the functions related to processing JunitXML and running tests out of the central runner into their own files. It also makes the header of the script a bit more friendly.

This PR is part of a series of refactoring PRs, split up into individual ones for ease of reviewing them.

### CODE REVIEW

File: testsuite/do_tests.sh.in [hunk 1]
```diff
@@ -116,239 +78,18 @@ time_cmd()
...
-}
 
-#
-# junit_close
-#
-junit_close ()
-{
-    if test "x${JUNIT_FILE}" = x ; then
-        bail_out 'junit_close: report file not open, call junit_open first!'
     fi
```
Comment (jougs) [inline testsuite/do_tests.sh.in:89]:
```suggestion
```

**Comment by jougs** [comment]
OK. It's unbroken again. You can start reviewing!

File: testsuite/run_test.sh [hunk 2]
```diff
@@ -0,0 +1,152 @@
+
```
[the remark this answers is not in the document]
Comment (clinssen) [reply testsuite/run_test.sh:1]: On second thought, it looks like this script is never run directly but only sourced. So perhaps leaving this line out (and leaving permissions -x) is the best thing to do. Possibly include a one-line comment indicating that this script is not intended to be run directly, but is used by ``run_tests.sh``.

**Comment by clinssen** [comment]
I performed a fresh install && installcheck, and everything worked out of the box.

File: testsuite/run_test.sh (hunk 2 above)
[the remark this answers is not in the document]
Comment (jougs) [reply testsuite/run_test.sh:1]: Done in 1ffaa0b.

File: testsuite/do_tests.sh.in [hunk 3]
```diff
@@ -1,21 +1,12 @@
 #!/bin/sh
 #
```
[the remark this answers is not in the document]
Comment (jougs) [reply testsuite/do_tests.sh.in:8]: Done in 1ffaa0b.

File: testsuite/run_test.sh (hunk 2 above)
[the remark this answers is not in the document]
Comment (jougs) [reply testsuite/run_test.sh:1]: I've added a short informational header in ee8b0f087, explaining the purpose of the helper scripts.

File: testsuite/run_test.sh [hunk 4]
```diff
@@ -0,0 +1,152 @@
...
+
+#
+# run_test script_name codes_success codes_failure
```
[the remark this answers is not in the document]
Comment (jougs) [reply testsuite/run_test.sh]: Fixed in e645e7c together with a fix for a type introduced in 1ffaa0b.

File: testsuite/junit_xml.sh [hunk 5]
```diff
@@ -0,0 +1,120 @@
+
```
[the remark this answers is not in the document]
Comment (jougs) [reply testsuite/junit_xml.sh:1]: Done in 1ffaa0b.

**Comment by jougs** [comment]
Thanks for the reviews. I've addressed all of your concerns in my latest commits.

**Pull request merged**